### PR TITLE
allow spaces in credit card number, with dynamic card number length r…

### PR DESCRIPTION
**Pull request by ScottDowne**
…estrictions

### CODE REVIEW

File: components/credit-card-info.jsx [hunk 1]
```diff
@@ -69,7 +70,7 @@ var CreditCardInfo = React.createClass({
   onCardInput: function(e) {
     var value = e.currentTarget.value;
     var state = this.state;
-    if (/^(\d)*$/.test(value)) {
+    if (/^(\d| )*$/.test(value) && value.replace(/ /g, "").length <= 16) {
```
Comment (ScottDowne) [inline components/credit-card-info.jsx]: Allow spaces, and make sure our length restriction ignores spaces.

File: components/credit-card-info.jsx [hunk 2]
```diff
@@ -31,6 +31,7 @@ var CreditCardInfo = React.createClass({
     if (this.props.error.number) {
       return false;
     }
+    cardNumber = cardNumber.replace(/ /g, "");
```
Comment (ScottDowne) [inline components/credit-card-info.jsx]: Make sure our card number validation ignores spaces for now.

**Comment by ScottDowne** [comment]
Just allows spaces, and stripe is fine with spaces.

It also makes sure our max length handles the added spaces or not.

r?

**Comment by alicoding** [comment]
r+

**Pull request merged**